File: Seccion 11 - Funciones en C++/Seccion_11_Dart/13_parametros_vector.dart

void cuadrado(List<int> vector, int tam) {
  for (int i = 0; i < tam; i++) {
    vector[i] = vector[i] * vector[i];
  }
}

void muestra(List<int> vector, int tam) {
  for (int i = 0; i < tam; i++) {
    print(vector[i]);
  }
}

void main() {

  const int TAM = 5;
  List<int> vector = [1, 2, 3, 4, 5];

  cuadrado(vector, TAM);
  muestra(vector, TAM);
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/13_parametros_vector.py

TAM = 5
vector = [1, 2, 3, 4, 5]

def cuadrado():
    global vector
    
    for i in range(TAM):
        vector[i] = vector[i] * vector[i]

def muestra():
    for i in range(TAM):
        print(vector[i])

cuadrado()
muestra()